### ISSUE TITLE
SAML

### ISSUE THREAD

**Issue by Firstyear** [kanidm/kanidm#2025]
Some applications only support SAML. It may be the case that we have to add SAML (even though it is a notoriously insecure and complex protocol). 

We should always guide people to use oauth2/oidc first, and then LDAP app passwords second, with SAML as an absolute last resort. If we end up in that situation, we may need to add it.

**Comment by yaleman** [comment]
SAML, like LDAP, seems like a necessary feature especially for Enterprise-style/scale estates. I've got a bunch of code which can work towards this, just ... some of the more XML'y bits were trouble at the time. It's been a while, so I might have a a chance of doing it these days though 😄

**Comment by Firstyear** [comment]
Yes, but unlike LDAP, SAML has huge security risks associated with it in both the implementation and clients. So I want to avoid this *as much as possible* and even if we do add it, it's going to be behind a flag of "enable-insecure-saml-provider" with an explanation of why users should choose literally anything else.

**Comment by yaleman** [comment]
oh, for sure.

**Comment by yaleman** [comment]
It's a shame there isn't a *good* alternative for this so we can just build an integration for it  (SimpleSAMLPHP and Keycloak need not apply)

**Comment by Firstyear** [comment]
Because SAML isn't good :)

**Comment by Mythra** [comment]
So 👀 , I've been using KanIDM for my personal homelab that connects with some other friends, I manage an instance of GitHub Enterprise Server. [GHES (unlike GHEC (github enterprise cloud)), only supports CAS, LDAP, and SAML](https://docs.github.com/en/enterprise-server@3.15/admin/managing-iam/understanding-iam-for-enterprises/about-identity-and-access-management#external-authentication).

Now this is a bummer for me, because I'd prefer to not have a separate LDAP app password (right now they already have to have a GitHub built in authentication username/password), I want to remove knowing a second password in an ideal world. Although GHES did just have a saml bypass in their latest release, and I would go on the record as being a SAML hater, given the GHES instance is only accessible over VPN behind a private key/oidc. I'd kind of like to explore the idea of using SAML + SCIM on this instance.

So has there been any work done on SAML so far in KanIDM? Would y'all be open to contributions if someone wanted to seriously explore this?

**Comment by yaleman** [comment]
No work yet, I'd be happy to see a plan/design for a rust SAML implementation.

**Comment by codering** [comment]
Has this requirement been scheduled yet?

**Comment by yaleman** [comment]
No

**Comment by Firstyear** [comment]
https://github.com/dexidp/dex/discussions/1884

Evidence of other providers removing SAML due to security concerns. At some point I think we should close this issue - openid federation is starting to be adopted and is likely a better path. People who need SAML have openid to SAML bridges they can use. 

I think the security risks and feature scope of implementing SAML is too great and we should commit to not supporting it.

**Comment by spprod35** [comment]
Just to follow up on @Firstyear's comment about OpenID Federation being a better path:
OpenID Federation 1.0 is now a Final Specification (published February 17, 2026).
https://openid.net/specs/openid-federation-1_0.html
This seems like the right moment to close this issue in favor of the OpenID Federation approach. The spec being finalized should help drive adoption and gives Kanidm a stable target to implement against when the time comes.
For reference, the related tracking issue appears to be #149 ("Implement trusts").
